cosine_similarity_model.py

```diff
@@ -0,0 +1,179 @@
+import json
+import numpy as np
+from sklearn.metrics.pairwise import cosine_similarity
+from pprint import pprint
+from collections import Counter
+import logging
+
+def setup_logger():
+    logger = logging.getLogger('logfile.log')
+    logger.setLevel(logging.INFO)
+    handler = logging.StreamHandler()
+    formatter = logging.Formatter('%(asctime)s - %(levelname)s - %(message)s')
+    handler.setFormatter(formatter)
+    logger.addHandler(handler)
+    return logger
+
+
+logger = setup_logger()
+
+
+def load_movies_from_json(file_path):
+    with open(file_path, "r", encoding="utf-8") as file:
+        data = json.load(file)
+    return data
+
+
+def extract_genres(movies):
+    all_genre_ids = set()
+    for movie in movies:
+        details = movie.get('detail', [])
+        for detail in details:
+            genres = detail.get('genres', [])
+            for genre in genres:
+                all_genre_ids.add(genre['id'])
+    return list(all_genre_ids)
+
+
+def extract_actors(movies):
+    all_actors = set()
+    for movie in movies:
+        credits = movie.get('credit', [])
+        for credit in credits:
+            cast = credit.get('cast', [])
+            for actor in cast:
+                all_actors.add(actor['name'])
+    return list(all_actors)
+
+
+def extract_keywords(movies):
+    all_keywords = set()
+    for movie in movies:
+        if 'keyword' in movie:
+            for keyword_group in movie['keyword']:
+                if 'keywords' in keyword_group:
+                    for keyword in keyword_group['keywords']:
+                        all_keywords.add(keyword['name'])
+    return list(all_keywords)
+
+
+def create_feature_vector(movie, all_genre_ids):
```

Comment:
use `CountVectorizer` from `sklearn.feature_extraction.text` to create the feature vector
simply create a "soup" from the extracted data and pass it to the vectorizer
```
soup = (keywords + cast + director + genres)
count = CountVectorizer(...)
count_matrix = count.fit_transform(soup)
```